src/System.Runtime.Serialization.Xml/tests/Performance/PerformanceTestsCommon.cs

```diff
@@ -0,0 +1,207 @@
+            }
+            return obj;
+        }
+
+        public static byte[] CreateByteArray(int size)
+        {
+            byte[] obj = new byte[size];
+            for (int i = 0; i < obj.Length; ++i)
+            {
+                obj[i] = (byte)(i % 256);
+            }
+            return obj;
+        }
+
+        public static Dictionary<int, string> CreateDictionaryOfIntString(int count)
+        {
+            Dictionary<int, string> dictOfIntString = new Dictionary<int, string>();
+            for (int i = 0; i < count; ++i)
+            {
+                dictOfIntString[i] = i.ToString();
+            }
+
+            return dictOfIntString;
+        }
+
+        public static List<int> CreateListOfInt(int count)
+        {
+            return Enumerable.Range(0, count).ToList();
+        }
+
+        public static object GetSerializationObject(TestType testType, int testSize)
+        {
+            Object obj = null;
+            switch (testType)
+            {
+                case TestType.ByteArray:
+                    obj = CreateByteArray(testSize);
+                    break;
+                case TestType.Dictionary:
+                    obj = CreateDictionaryOfIntString(testSize);
+                    break;
+                case TestType.ISerializable:
+                    obj = new ClassImplementingIXmlSerialiable() { StringValue = "Hello world" };
+                    break;
+                case TestType.ListOfInt:
+                    obj = CreateListOfInt(testSize);
+                    break;
+                case TestType.SimpleType:
+                    obj = CreateObject(testSize, 0, -1, 7, 2);
+                    break;
+                case TestType.String:
+                    obj = new string('k', testSize);
+                    break;
+                case TestType.XmlElement:
+                    XmlDocument xmlDoc = new XmlDocument();
+                    xmlDoc.LoadXml(@"<html></html>");
+                    XmlElement xmlElement = xmlDoc.CreateElement("Element");
+                    xmlElement.InnerText = "Element innertext";
+                    obj = xmlElement;
+                    break;
+            }
+            return obj;
+        }
+
+        #endregion
+
+        #region Methods to run serialization performance tests
+
+        public static void RunSerializationPerformanceTest(int iterations, SerializerType serializerType, TestType testType, int testSize, SerializerFactory serializerFactory)
+        {
+            var obj = GetSerializationObject(testType, testSize);
+
+            var serializer = serializerFactory.GetSerializer(serializerType);
+            serializer.Init(obj);
+
+            using (var stream = new MemoryStream())
+            {
+                foreach (var iteration in Benchmark.Iterations)
+                {
+                    using (iteration.StartMeasurement())
+                    {
+                        for (int i = 0; i < iterations; i++)
```

Comment:
The word iteration is being overloaded. Benchmark.Iterations is referring to an iteration of running the Benchmark, whereas the iterations here is referring to the count of the number of times your are serializing. It's a little confusing.